src/theme-context-provider.tsx

```diff
@@ -70,16 +80,21 @@ const useDefaultHrefDecorator = () => {
     return (href: string) => href;
 };
 
-const ThemeContextProvider: React.FC<Props> = ({theme, children}) => {
+const ThemeContextProvider: React.FC<Props> = ({theme, children, providerId}) => {
+    const [instanceId] = React.useState(() => {
+        if (providerId) {
+            return providerId;
+        } else {
+            return isServerSide() ? 0 : nextJssInstanceId++;
+        }
+    });
```

Comment:
do you need the useState? why don't you do this logic inside the useMemo?

Reply:
I don't want the useMemo to depend on the prop. Once the value is set I don't want to allow it to change. A better option would be to use a ref but useRef doesn't accept a callback